vumi/persist/redis_base.py

```diff
@@ -55,14 +55,24 @@ def __new__(meta, classname, bases, class_dict):
         return type.__new__(meta, classname, bases, new_class_dict)
 
 
+class ClientProxy(object):
+    def __init__(self, client):
+        self.client = client
+
+
 class Manager(object):
 
     __metaclass__ = CallMakerMetaclass
 
-    def __init__(self, client, config, key_prefix, key_separator=None):
+    def __init__(
+            self, client, config, key_prefix, key_separator=None,
+            client_proxy=None):
         if key_separator is None:
             key_separator = ':'
-        self._client = client
+        if client_proxy is None:
+            self._client_proxy = ClientProxy(client)
+        else:
+            self._client_proxy = client_proxy
```

Comment:
This ensures that submanagers use the same client, while root managers create a client proxy. I'm not sure of a better way of doing this without changing the API. Ideas?